examples/citi_bike/online_lp/citi_bike_lp.py

```diff
@@ -0,0 +1,187 @@
+import math
+from typing import List, Tuple
+
+import numpy as np
+from pulp import LpProblem, LpVariable, LpMaximize, lpSum, LpStatus, GLPK
+
+from maro.utils import DottableDict
+
+class LP():
+    def __init__(
+        self, num_station: int, num_neighbor: int,
+        station_capacity: List[int], station_neighbor_list: List[List[int]],
+        decision_interval: int, config: DottableDict
+    ):
+        """A simple Linear Programming formulation for solving the bike repositioning problem.
+
+        Args:
+            num_station (int): Number of stations in current topology.
+            num_neighbor (int): Number of neighbors that needed to consider when repositioning.
+            station_capacity (List[int]): The capacity of each station. Length: num_station.
+            station_neighbor_list (List[List[int]]): The neighbor station of each station. Size: (num_station, num_station).
+            decision_interval (int): The time interval of the environment to check and trigger a decision_event.
+            config (DottableDict): LP configuration. Including the configuration for
+                problem formulation, objectives, solution application.
+        """
+        self._num_station = num_station
+        self._num_neighbor = num_neighbor
+        self._station_capacity = station_capacity
+        self._station_neighbor_list = station_neighbor_list
+        self._decision_interval = decision_interval
+
+        self._plan_window_size = config.plan_window_size
+        self._apply_buffer_size = config.apply_buffer_size
+        self._safety_inventory_limit = config.safety_inventory_limit
+
+        self._fulfillment_time_decay_factor = config.objective.fulfillment_time_decay_factor
+        self._transfer_cost_factor = config.objective.transfer_cost_factor
+        self._safety_inventory_reward_factor = config.objective.safety_inventory_reward_factor
+
+        self._num_tick = math.ceil(self._plan_window_size / self._decision_interval)
+        self._last_start_tick = -1
+
+    def _init_variables(self, init_inventory: np.ndarray):
+        self._inventory = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._safety_inventory = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._fulfillment = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
+        self._transfer = np.zeros((self._num_tick, self._num_station, self._num_neighbor), dtype=np.int32).tolist()
+
+        # The intermediate variables for clearer constraints.
+        self._transfer_from = np.zeros((self._num_tick, self._num_station), dtype=np.int32).tolist()
```

Comment:
np.uint16 is enough for this case?
https://numpy.org/doc/stable/user/basics.types.html